File: backend/agentflow/schemas/log.py
"""Log / marketplace request/response schemas."""
from __future__ import annotations

from pydantic import BaseModel


class NodeRunLogOut(BaseModel):
    id: int
    run_id: str
    node_id: str
    node_type: str
    status: str
    inputs: dict = {}
    outputs: dict = {}
    error: str | None = None
    elapsed_ms: int = 0
    created_at: str | None = None


class WorkflowRunOut(BaseModel):
    run_id: str
    workflow_id: str
    mode: str
    status: str
    inputs: dict = {}
    outputs: dict = {}
    error: str | None = None
    total_ms: int = 0
    created_at: str | None = None


class PublishTemplateRequest(BaseModel):
    category: str = "通用"


class ImportTemplateRequest(BaseModel):
    template_id: str

File: frontend/src/types/log.ts
// Debug logs — mirrors backend schemas/log.py
export interface LLMDebug {
  model?: string
  system?: string
  prompt?: string
  raw_response?: string
  resolved_inputs?: Record<string, any>
  llm_available?: boolean
}

export interface NodeRunLog {
  id?: number
  run_id: string
  workflow_id?: string
  node_id: string
  node_type: string
  status: 'running' | 'succeeded' | 'failed'
  inputs: Record<string, any>
  outputs: Record<string, any>
  debug?: LLMDebug & Record<string, any>
  error?: string | null
  elapsed_ms: number
  created_at?: string
}

export interface WorkflowRun {
  run_id: string
  workflow_id: string
  mode: string
  status: string
  inputs: Record<string, any>
  outputs: Record<string, any>
  error?: string | null
  total_ms: number
  created_at?: string
}
